gef.py

```diff
@@ -10600,26 +10610,26 @@ class GefHelpCommand(gdb.Command):
     _cmdline_ = "gef help"
     _syntax_  = _cmdline_
 
-    def __init__(self, commands, *args, **kwargs):
+    def __init__(self, commands: List[Tuple[str, Any, Any]], *args, **kwargs) -> None:
         super().__init__(self._cmdline_, gdb.COMMAND_SUPPORT, gdb.COMPLETE_NONE, False)
         self.docs = []
         self.generate_help(commands)
         self.refresh()
         return
 
-    def invoke(self, args, from_tty):
+    def invoke(self, args, from_tty) -> None:
         self.dont_repeat()
         gef_print(titlify("GEF - GDB Enhanced Features"))
         gef_print(self.__doc__)
         return
 
-    def generate_help(self, commands):
+    def generate_help(self, commands: List[Tuple[str, Any, Any]]) -> None:
```

Comment:
I tracked this through:


```python
__commands__ -> list of classes
self.commands = [(x._cmdline_, x) for x in __commands__]
for cmd, class_name in self.commands:
        loaded_commands.append((cmd, class_name, class_name())
GetHelpCommand(loaded_commands)
__init__(commands=loaded_commands)
self.generate_help(commands)
for command in commands:
        self.add_command_to_doc(command)

cmd, class_name, _ = command
```

So basically: It should be `List[Tuple[str, ClassVar, instance]]`, so really `List[Tuple[str, ClassVar, any]]`, though I am not sure if that type exists in 3.6. Also, we keep calling it `class_name` when really it should be `cls` or `class_object` and `class_object` should be `class_instance`.

I could fix this up for you if it's too annoying.

Reply:
It should be `Type`, not `ClassVar`.